src/mscorlib/src/System/Guid.cs

```diff
@@ -1285,35 +1285,40 @@ public String ToString(String format)
             return ToString(format, null);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static char HexToChar(int a)
         {
             a = a & 0xf;
             return (char)((a > 9) ? a - 10 + 0x61 : a + 0x30);
         }
 
-        unsafe private static int HexsToChars(char* guidChars, int offset, int a, int b)
+        unsafe private static int HexsToChars(char* guidChars, int a, int b)
         {
-            return HexsToChars(guidChars, offset, a, b, false);
```

Comment:
With the current inline model and without `AggressiveInlining`, you are treading in a narrow space of IL sizes. 16 bytes of IL or less will always inline, 33 bytes of IL or more will never inline, regardless of the number or kinds of args.

In between it depends on a cost/benefit analysis, and since there's both extra cost and extra benefit with more args, so there's no fixed rule as to when one will prevail over the other. 

How C# translates to IL is not always obvious so from C# this can appear a bit mysterious. 

Not sure the cost/benefit model is well motivated so won't try and drill into what it does. But if you're bored someday and have a method whose body is between 16 and 32 bytes of IL, try adding a bunch of unused arguments and see what happens.